pkg/target/kusttarget_test.go

```diff
@@ -336,3 +336,171 @@ bases:
+			Name:       "chill"},
+		FieldRef: types.FieldSelector{FieldPath: "metadata.name"},
+	},
+	{
+		Name: "FRUIT",
+		ObjRef: types.Target{
+			Gvk:  gvk.Gvk{Kind: "Service"},
+			Name: "apple"},
+		FieldRef: types.FieldSelector{FieldPath: "metadata.name"},
+	},
+	{
+		Name: "VEGETABLE",
+		ObjRef: types.Target{
+			Gvk:  gvk.Gvk{Kind: "Leafy"},
+			Name: "kale"},
+		FieldRef: types.FieldSelector{FieldPath: "metadata.name"},
+	},
+}
+
+func TestGetAllVarsSimple(t *testing.T) {
+	ldr := loadertest.NewFakeLoader(
+		"/app")
+	write(t, ldr, "/app", `
+vars:
+  - name: HOT
+    objref:
+      kind: Service
+      name: hotness
+      apiVersion: v7
+    fieldref:
+      fieldpath: rumpelstiltskin
+  - name: COLD
+    objref:
+      kind: Service
+      name: chill
+      apiVersion: v300
+`)
+	vars, err := makeKustTarget(t, ldr).getAllVars()
+	if err != nil {
+		t.Fatalf("Err: %v", err)
+	}
+	if len(vars) != 2 {
+		t.Fatalf("unexpected size %d", len(vars))
+	}
+	for i, v := range someVars[:2] {
+		if !reflect.DeepEqual(vars[i], v) {
+			t.Fatalf("unexpected var[%d]:\n  %v\n  %v", i, vars[i], v)
+		}
+	}
+}
+
+func TestGetAllVarsNested(t *testing.T) {
+	ldr := loadertest.NewFakeLoader(
+		"/app/overlays/o2")
+	write(t, ldr, "/app/base", `
+vars:
+  - name: HOT
+    objref:
+      kind: Service
+      name: hotness
+      apiVersion: v7
+    fieldref:
+      fieldpath: rumpelstiltskin
+  - name: COLD
+    objref:
+      kind: Service
+      name: chill
+      apiVersion: v300
+`)
+	write(t, ldr, "/app/overlays/o1", `
+vars:
+  - name: FRUIT
+    objref:
+      kind: Service
+      name: apple
+bases:
+- ../../base
+`)
+	write(t, ldr, "/app/overlays/o2", `
+vars:
+  - name: VEGETABLE
+    objref:
+      kind: Leafy
+      name: kale
+bases:
+- ../o1
+`)
+	vars, err := makeKustTarget(t, ldr).getAllVars()
+	if err != nil {
+		t.Fatalf("Err: %v", err)
+	}
+	if len(vars) != 4 {
+		t.Fatalf("unexpected size %d", len(vars))
+	}
+	for i, v := range someVars {
+		if !reflect.DeepEqual(vars[i], v) {
+			t.Fatalf("unexpected var[%d]:\n  %v\n  %v", i, vars[i], v)
+		}
+	}
+}
+
+func TestGetAllVarsCollisionsOkay_PleaseFix(t *testing.T) {
+	ldr := loadertest.NewFakeLoader(
+		"/app/overlays/o2")
+	write(t, ldr, "/app/base", `
+vars:
+  - name: HOT
+    objref:
+      kind: Service
+      name: hotness
+      apiVersion: v7
+    fieldref:
+      fieldpath: rumpelstiltskin
+  - name: COLD
+    objref:
+      kind: Service
+      name: chill
+      apiVersion: v300
+`)
+	write(t, ldr, "/app/overlays/o1", `
+vars:
+  - name: HOT
+    objref:
+      kind: Service
+      name: apple
+bases:
+- ../../base
+`)
+	write(t, ldr, "/app/overlays/o2", `
+vars:
+  - name: VEGETABLE
+    objref:
+      kind: Leafy
+      name: kale
+bases:
+- ../o1
+`)
+	vars, err := makeKustTarget(t, ldr).getAllVars()
+	if err != nil {
+		t.Fatalf("Err: %v", err)
+	}
+	if len(vars) != 4 {
+		t.Fatalf("unexpected size %d", len(vars))
+	}
+	v2 := someVars[2]
+	v2.Name = "HOT"
```

Comment:
What is the purpose of this test?  Overlay o1 and base both have env `HOT`. Do we pick up the overlay one or base one? Is this a conflict?